src/System.Numerics.Vectors/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1786,64 +1787,102 @@ public static Matrix4x4 Transpose(Matrix4x4 matrix)
         /// <param name="matrix2">The second source matrix.</param>
         /// <param name="amount">The relative weight of the second source matrix.</param>
         /// <returns>The interpolated matrix.</returns>
-        public static Matrix4x4 Lerp(Matrix4x4 matrix1, Matrix4x4 matrix2, float amount)
+        public static unsafe Matrix4x4 Lerp(Matrix4x4 matrix1, Matrix4x4 matrix2, float amount)
         {
-            Matrix4x4 result;
+            if (Sse.IsSupported)
+            {
+                var amountVec = Sse.SetAllVector128(amount);
 
-            // First row
-            result.M11 = matrix1.M11 + (matrix2.M11 - matrix1.M11) * amount;
-            result.M12 = matrix1.M12 + (matrix2.M12 - matrix1.M12) * amount;
-            result.M13 = matrix1.M13 + (matrix2.M13 - matrix1.M13) * amount;
-            result.M14 = matrix1.M14 + (matrix2.M14 - matrix1.M14) * amount;
+                var m1Row = Sse.LoadVector128(&matrix1.M11);
```

Comment:
Do we need to have `[StructLayout(LayoutKind.Sequential)]` on the `Matrix4x4` type? That way we are assured the members will be laid out in the assumed order?

Reply:
is not the default layout sequential? 
Also there is an opinion that a better layout for matrices is column-major order (this one is row-major) for popular matrix operations in game development (but of course it's a breaking change and doesn't match DirectX)

Reply:
The Roslyn Compiler emits `StructLayout=Sequential` by default for struct types. However, last I checked there wasn't anything enforcing this behavior in the language spec.

It would probably be a good idea to explicitly mark the numeric types as Sequential (as we do with types like `Single` and `Int32`).